query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java

```diff
@@ -276,6 +283,28 @@ private String convertOperatorToString(Operator operator) {
     }
   }
 
+  private String handleLikeOperator(Expression expression) {
+    Optional<String> logicalColumnName = getLogicalColumnName(expression);
+    if (logicalColumnName.isPresent()
+        && isSimpleAttributeExpression(expression)
+        && viewDefinition.hasTextIndex(logicalColumnName.get())) {
+      return TEXT_MATCH_OPERATOR;
+    }
+    return REGEX_OPERATOR;
+  }
+
+  private Expression prefixValueForLikeOperator(String likeOperatorStr, Expression rhsExpression) {
```

Comment:
For TEXT_MATCH regex query needs string to be prefixed and suffixed with '/'.

- https://docs.pinot.apache.org/basics/indexing/text-search-support